src/transformStateWithClones.js

```diff
@@ -65,7 +65,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  let copyState = { ...state };
+  const result = [];
+
+  for (const key of transforms) {
+    switch (key.operation) {
+      case 'addProperties':
+        Object.assign(copyState, key.properties);
+        break;
+
+      case 'removeProperties':
+        for (const i of key.properties) {
```

Comment:
Don't use 'i' in such cycle. Use meaningful variable name. 'i' is appropriate for cycle with numbers,  like
for (const i = 0; i < someString.length; i++)